contracts/account-history-contract/src/action/query/get_estaking_rewards.rs

```diff
@@ -17,12 +17,20 @@ pub fn get_estaking_rewards(
 
     let response = querier.get_estaking_rewards(address).unwrap_or_default();
 
-    let fiat_coins = response
-        .to_dec_coin_values(&querier, &generator.metadata.usdc_denom)
-        .unwrap_or_default();
+    let fiat_coins = response.to_dec_coin_values(&querier, &generator.metadata.usdc_denom);
+
+    match fiat_coins {
+        Err(e) => {
+            return Err(StdError::generic_err(format!(
+                "Failed to convert to DecCoinValue {}",
+                e
+            )))
+        }
+        _ => {}
+    }
 
     Ok(GetEstakingRewardsResponse {
-        rewards: fiat_coins,
+        rewards: fiat_coins.unwrap_or_default(),
```

Comment:
```suggestion
        rewards: fiat_coins,
```

because above `?` will handle the error case